quiz/quiz.go

```diff
@@ -0,0 +1,182 @@
+// This package implements the quiz, with a variation on part 2.  
+// Instead of having an timer for the entire test, it handles a timer 
+// for each question.  This was more challenging, since characters 
+// entered for previous questions had to be properly discarded
+// to get the desired behavior.
+
+package main
+
+import (
+	"bufio"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+	"time"
+)
+
+type QuizItem struct {
+	Question string
+	Answer   string
+}
+
+type CharWithTime struct {
+	char string
+	time time.Time
+}
+
+type ErrTimeout struct {
```

Comment:
This should be `TimeoutErr` or `TimeoutError` (or non exported since it is in the main file). The naming Err<something> is typicaly for spicific errors e.g. `var ErrTimeout = errors.New("timeout error")`.

On another note, the existence of this type suggests that something weird is happening and that you probably have a lot of problems that a `context.Context` could solve.

Reply:
Thanks for the feedback on naming custom errors. 
Instead of using a custom error type here, I should have just used the ,ok idiom. 
I'm not sure how context.Context would help here, but I've used context cancellation with a go web server in a tutorial that I've gone through at home, so I'm familar with it.